### PR TITLE
fix(Scripts/Naxxramas): Add post-Gothik Four Horsemen introduction

### PR DESCRIPTION
**Pull request by sogladev**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

add the RP

## Issues Addressed:

- Progresses https://github.com/azerothcore/azerothcore-wotlk/issues/4663

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [x] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

horsesniffer

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [x] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

go to gothik
1. `.go c id 16060`
2. `.die` gothik
3. wait 10 seconds for RP to start

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Northrend/Naxxramas/boss_four_horsemen.cpp [hunk 1]
```diff
@@ -75,9 +75,9 @@ enum Misc
 
 enum FourHorsemen
 {
-    SAY_AGGRO                           = 0,
-    SAY_TAUNT                           = 1,
-    SAY_SPECIAL                         = 2,
+    // SAY_AGGRO                           = 0,
+    // SAY_TAUNT                           = 1,
+    // SAY_SPECIAL                         = 2,
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Naxxramas/boss_four_horsemen.cpp:80]: if you dont use them, delete them

**Review by Kitzunu — APPROVED** [review]

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp [hunk 2]
```diff
@@ -802,10 +803,33 @@ class spell_gothik_shadow_bolt_volley : public SpellScript
...
+    }
+};
+
 void AddSC_boss_gothik()
 {
     new boss_gothik();
     new npc_boss_gothik_minion();
     new npc_gothik_trigger();
     RegisterSpellScript(spell_gothik_shadow_bolt_volley);
+    new at_gothik_entrance();
```
Comment (sudlud) [inline src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:834]: Can new register method be used here?

File: src/server/scripts/Northrend/Naxxramas/instance_naxxramas.cpp [hunk 3]
```diff
@@ -1227,9 +1280,33 @@ class at_naxxramas_hub_portal : public AreaTriggerScript
...
+        return false;
+    }
+};
+
 void AddSC_instance_naxxramas()
 {
     new instance_naxxramas();
     new boss_naxxramas_misc();
     new at_naxxramas_hub_portal();
+    new at_deathknight_wing_entrance();
```
Comment (sudlud) [inline src/server/scripts/Northrend/Naxxramas/instance_naxxramas.cpp:1311]: same here

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp (hunk 2 above)
Comment (sogladev) [reply src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:834]: Register macros for Areatriggers do not exist yet.

[I implemented something like this, but out of scope for this PR](https://github.com/azerothcore/azerothcore-wotlk/commit/fad198a445c6d91d4f368afffab679b48ffbb204#diff-6396c45ad9207c72c30ee43ea53d6bc708bb7dc4a7a2b5e3414b4a301abcbccd)
I 
```
struct at_deathknight_wing_entrance : OnlyOnceAreaTriggerScript
{
    PrepareOnlyOnceAreaTriggerScript(at_deathknight_wing_entrance);

    bool _OnTrigger(Player* player, AreaTrigger const* /*at*/) override
    {
        ...
    }
};
```

versus

```
class at_deathknight_wing_entrance : public OnlyOnceAreaTriggerScript
{
public:
    at_deathknight_wing_entrance() : OnlyOnceAreaTriggerScript("at_deathknight_wing_entrance") { }

    bool _OnTrigger(Player* player, AreaTrigger const* /*at*/) override
    {
        ...
    }
};
```

**Comment by TheSCREWEDSoftware** [comment]
@sogladev i'm confused on the order of testing this PR.

**Comment by sogladev** [comment]
> @sogladev i'm confused on the order of testing this PR.

Sorry, I wrote them out of order. I updated the test section, so it should be in the order players clear the Military Wing

**Comment by TheSCREWEDSoftware** [comment]
@sogladev I might be doing something. I'm not teleporting (running and killing) because there are triggers area so prefer to not miss them.

https://youtu.be/dzxmM1mER80?si=KdC4RrQ_zE-fhXUW

**Comment by sogladev** [comment]
> @sogladev I might be doing something. I'm not teleporting (running and killing) because there are triggers area so prefer to not miss them.
> 
> https://youtu.be/dzxmM1mER80?si=KdC4RrQ_zE-fhXUW

`.debug areatrigger` will print if you hit an areatrigger

I see Zeliek didn't print the 2nd message, it should print 30 seconds after entering Gothik's room.  :( will have to revisit. Thanks for testing

**Comment by TheSCREWEDSoftware** [comment]
Ping me again when you need another testing. I don't mind testing PRs, people just have to tell me or ping me :)

**Comment by sogladev** [comment]
I removed the area trigger dialogue because I found some unknowns. The 1st taunt is random from any horseman and I'll have to investigate what causes the 2nd "taunt" message and edit the creature_texts. 

@TheSCREWEDSoftware could you test this again? Just the post-Gothik Four Horsemen introduction

**Comment by TheSCREWEDSoftware** [comment]
> I removed the area trigger dialogue because I found some unknowns. The 1st taunt is random from any horseman and I'll have to investigate what causes the 2nd "taunt" message and edit the creature_texts.
> 
> @TheSCREWEDSoftware could you test this again? Just the post-Gothik Four Horsemen introduction

So the 30 seconds of the gothink and his dialogue when death or just the way to the horseman (from gothink) and the fight

**Comment by sogladev** [comment]
> > I removed the area trigger dialogue because I found some unknowns. The 1st taunt is random from any horseman and I'll have to investigate what causes the 2nd "taunt" message and edit the creature_texts.
> > @TheSCREWEDSoftware could you test this again? Just the post-Gothik Four Horsemen introduction
> 
> So the 30 seconds of the gothink and his dialogue when death or just the way to the horseman (from gothink) and the fight

When Gothik dies, there should be a dialogue from the horsemen. They should match the pictures in the linked issue  labelled "Part 1" and "Part 2" https://github.com/azerothcore/azerothcore-wotlk/issues/4663

**Comment by TheSCREWEDSoftware** [comment]
@sogladev https://www.youtube.com/watch?v=3tjrFFh8ysM&ab_channel=TheSCREWEDSoftware (this one took 4ever to upload idk why) same size as the others...
